src/algorithms/libs/galileo_e5_signal_processing.cc

```diff
@@ -123,3 +126,100 @@ void galileo_e5_a_code_gen_complex_sampled(own::span<std::complex<float>> _dest,
             _dest[(i + delay) % _samplesPerCode] = _code[i];
         }
 }
+
+
+void galileo_e5_b_code_gen_complex_primary(own::span<std::complex<float>> _dest,
+    int32_t _prn,
+    const std::array<char, 4>& _Signal)  // to differentiate between E5a and E5b signal
```

Comment:
Please make array of size 3. There is no possible confusion with E5a, since the name of the function is different. Keping the same interface will make things easier.